three-cards-divination/src/test/java/com/cyber/mysticism/tarot/service/ThreeCardsDivinationServiceTest.java

```diff
@@ -1,63 +1,92 @@
 package com.cyber.mysticism.tarot.service;
 
 import com.cyber.mysticism.tarot.json.Card;
+import com.cyber.mysticism.tarot.model.TarotUser;
 import com.cyber.mysticism.tarot.repository.TarotDeckRepository;
+import com.cyber.mysticism.tarot.repository.UserRepository;
 import com.cyber.mysticism.tarot.service.exceptions.DivinationException;
+import com.cyber.mysticism.tarot.service.exceptions.UserNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.mock.mockito.MockBean;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
 import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
 
-@SpringBootTest
+@ExtendWith(MockitoExtension.class)
 class ThreeCardsDivinationServiceTest {
 
-    @Autowired
-    private ThreeCardsDivinationService service;
-    @MockBean
+    @Mock
     private TarotDeckRepository tarotDeckRepository;
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private TarotUser tarotUser;
+    @InjectMocks
+    private ThreeCardsDivinationService service;
+
+    @BeforeEach
+    void setup() {
+        when(userRepository.findById("user")).thenReturn(Optional.of(tarotUser));
+        when(userRepository.findByEmail("email")).thenReturn(Optional.of(tarotUser));
+    }
```

Comment:
**Suggestion:** The `setup` method is setting up mocks for every test method in the class. However, not all test methods require these mocks. Consider moving the mock setup to the individual test methods that require them. This will make the tests more isolated and easier to understand. [maintainability]
```suggestion
    // Remove the setup method and move the mock setup to the individual test methods
```